z3c/dependencychecker/main.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+from z3c.dependencychecker.dependencychecker import logger
+from z3c.dependencychecker.dependencychecker import parse_command_line
+from z3c.dependencychecker.dependencychecker import set_log_level
+
+import os
+import pkg_resources
+import sys
+
+
+def main():
+    options, args = parse_command_line()
+    set_log_level(options.verbose)
+    path = get_path(args)
+    get_package_dependencies(path)
+
+
+def get_path(args):
+    """Get path from given arguments or use """
+    path = os.getcwd()
+    if len(args) > 0:
+        path = os.path.abspath(args[0])
+        if not os.path.isdir(path):
+            logger.info('Given path does not exist: {0}'.format(args[0]))
+            sys.exit(1)
+    logger.info('path used: %s', path)
+    return path
+
+
+def get_package_dependencies(path):
+    if not has_setup_py_file(path):
+        logger.info('No setup.py file found! on {0}'.format(path))
+        sys.exit(1)
+
+    new_path, filename = get_egg_info_folder(path)
+    if not new_path:
+        logger.info('No egg-info folder found! on {0}'.format(new_path))
+        sys.exit(1)
+    path = new_path
+    package_name = filename[:-len('.egg-info')]
+
+    working_set = pkg_resources.WorkingSet()
+    working_set.add_entry(path)
+    new_requirement = pkg_resources.Requirement.parse(package_name)
+    distribution = working_set.find(new_requirement)
+    dependencies = distribution.requires()
+
+    extras_list = {}
+    for extra in distribution.extras:
+        all_with_extra = distribution.requires(extras=(extra, ))
+        only_extra = set(all_with_extra) - set(dependencies)
+        extras_list[extra] = only_extra
+
+    from pprint import pprint
+    pprint(dependencies)
+    pprint(extras_list)
+
+
+def has_setup_py_file(path):
+    return bool([x for x in os.listdir(path) if x == 'setup.py'])
+
+
+def get_egg_info_folder(path):
+    def get_egg_folder(test_path):
+        return [
+            x
+            for x in os.listdir(test_path)
+            if x.endswith('.egg-info')
+        ]
+
+    egg_info = get_egg_folder(path)
+    if not egg_info:
+        logger.info('Try to append src/ to the current path %s', path)
+        path = os.path.join(path, 'src')
+        egg_info = get_egg_folder(path)
+
+    if egg_info:
+        return path, egg_info[0]
```

Comment:
You use it as `new_path, filename` earlier up in the file. It is more readable if you pick one/two good explicit name(s) and use the same in both places.

Reply:
Yup, good idea 👍